report/renderer/performance-category-renderer.js

```diff
@@ -146,18 +92,50 @@ export class PerformanceCategoryRenderer extends CategoryRenderer {
   }
 
   /**
-   * For performance, audits with no group should be a diagnostic or opportunity.
-   * The audit details type will determine which of the two groups an audit is in.
+   * Returns true if the audit is a general performance insight (i.e. not a metric or hidden audit).
    *
    * @param {LH.ReportResult.AuditRef} audit
-   * @return {'load-opportunity'|'diagnostic'|null}
+   * @return {boolean}
    */
-  _classifyPerformanceAudit(audit) {
-    if (audit.group) return null;
-    if (audit.result.details?.overallSavingsMs !== undefined) {
-      return 'load-opportunity';
+  _isPerformanceInsight(audit) {
+    return !audit.group;
+  }
+
+  /**
+   * @param {LH.ReportResult.AuditRef} audit
+   * @param {LH.ReportResult.AuditRef[]} metricAudits
+   * @return {{overallImpact: number, overallLinearImpact: number}}
+   */
+  overallImpact(audit, metricAudits) {
+    if (!audit.result.metricSavings) {
+      return {overallImpact: 0, overallLinearImpact: 0};
+    }
+
+    let overallImpact = 0;
+    let overallLinearImpact = 0;
+    for (const [k, savings] of Object.entries(audit.result.metricSavings)) {
+      // Get metric savings for individual audit.
+      if (savings === undefined) continue;
+
+      // Get the metric data.
+      const mAudit = metricAudits.find(audit => audit.acronym === k);
+      if (!mAudit) continue;
+      if (mAudit.result.score === null) continue;
+
+      const mValue = mAudit.result.numericValue;
+      if (!mValue) continue;
+
+      const scoringOptions = mAudit.result.scoringOptions;
+      if (!scoringOptions) continue;
+
+      const newMetricScore = Util.computeLogNormalScore(scoringOptions, mValue - savings);
+      const linearImpact = savings / mValue * mAudit.weight;
+      overallLinearImpact += linearImpact;
```

Comment:
nit: we can actually compute the linear impact even if `scoringOptions` is missing. WDYT about moving these two lines above L128?

Reply:
sgtm!